### ISSUE TITLE
rename function for VOLUME objects missing?

### ISSUE THREAD

**Issue by Gaiiden** [KSP-KOS/KOS#1585]
I really like how I don't have to use SWITCH anymore and can just use COPY, CORE and ARCHIVE to move files around and stuff and not worry about what volume I'm currently on. But I can't rename files without using SWITCH to set the volume I want to rename the file on. It seems a strange omission

**Comment by hvacengi** [comment]
It probably isn't that hard to add the rename option.  Which syntax specifically are you talking about though.  We're at a point where we tend to add new features as suffixes, rather than the "particular syntax".  There is no suffix that allows copying, or renaming.  Are you referring to the following copy syntax?
```
copy myfile.ks to archive.
```
That syntax actually always accepted a volume identifier it was just:
```
copy myfile.ks from 0.
```

If you want to be able to do something like:
```
rename file myfile.ks to yourfile.ks on archive.
```
that presents a bit more of an issue in the parsing of the command, but is still doable.  I'm not much of a fan of this syntax, because it moves the "volume" parameter to be the last parameter.  It also means that the parameters are handled using magic in the compiler since it is the same thing as calling 
```
rename("file", "myfile.ks", "yourfile.ks", archive).
```

My prefered syntax for something like this would be:
```
archive:rename("myfile.ks", "yourfile.ks").
core:volume:rename("myfile.ks", "yourfile.ks").
```
Copy syntax on suffixes would be a little more complicated, but not by that much.
```
set file to core:volume:open("myfile.ks").
archive:copy(file, "yourfile.ks").
```
We would just need the file to retain it's volume information (and since it's a "Volume" file, that's probably a good idea).  Via this syntax we might even support copying files within a single volume.

@tomekpiotrowski as the file management guy, what do you think?  Is any of this covered by your work to get sub-directories working?

**Comment by tomekpiotrowski** [comment]
With subdirectories you'll be able to work with full paths so something like this will be possible out of the box:

```
rename "archive:/somefile.ks" to "othervolume:/othername".
```

This should make all file-related commands a lot more flexible and solve this problem as a side effect.

**Comment by tomekpiotrowski** [comment]
This will be implemented in #1567 and probably released with 1.0. I think I'll close this, we already have too many issue open ;)
